tests/pytests/unit/grains/test_core.py

```diff
@@ -1107,25 +1116,28 @@ def test__windows_os_release_grain():
         "Windows Embedded Standard 2009": "2009",
         "Windows XP Embedded": "XP",
     }
-    for caption in embedded_versions:
-        version = core._windows_os_release_grain(caption, 1)
-        assert version == embedded_versions[caption]
+    for caption, expected_version in embedded_versions.items():
+        with subtests.test(caption):
+            version = core._windows_os_release_grain(caption, 1)
+            assert version == expected_version
 
     # Special Cases
     # Windows Embedded Standard is Windows 7
     caption = "Windows Embedded Standard"
-    with patch("platform.release", MagicMock(return_value="7")):
-        version = core._windows_os_release_grain(caption, 1)
-        assert version == "7"
+    with subtests.test(caption):
+        with patch("platform.release", MagicMock(return_value="7")):
```

Comment:
**suggestion** similar simplification here - no need to create a new MagicMock, just `, return_value=7)`